src/WebJobs.Script/Workers/Rpc/RpcInitializationService.cs

```diff
@@ -127,79 +98,28 @@ internal async Task InitializeRpcServerAsync()
             }
             catch (Exception grpcInitEx)
             {
-                var hostInitEx = new HostInitializationException($"Failed to start Rpc Server. Check if your app is hitting connection limits.", grpcInitEx);
+                throw new HostInitializationException($"Failed to start Rpc Server. Check if your app is hitting connection limits.", grpcInitEx);
             }
         }
 
         internal Task InitializeChannelsAsync()
         {
-            // TODO: Remove special casing when resolving https://github.com/Azure/azure-functions-host/issues/4534
-            if (ShouldStartAsPlaceholderPool())
-            {
-                return _webHostRpcWorkerChannelManager.InitializeChannelAsync(_workerRuntime);
-            }
-            else if (ShouldStartStandbyPlaceholderChannels())
+            if (_placeholderLanguageWorkersList == null)
             {
-                return InitializePlaceholderChannelsAsync();
+                throw new ArgumentNullException(nameof(_placeholderLanguageWorkersList));
             }
 
-            return InitializeWebHostRuntimeChannelsAsync();
-        }
-
-        private Task InitializePlaceholderChannelsAsync()
-        {
-            if (_environment.IsLinuxAzureManagedHosting())
+            if (_environment.IsPlaceholderModeEnabled())
```

Comment:
So is this whole method now a no-op unless in placeholder mode? That's now how it used to work, is it? I see it used to eventually call into `InitializeWebHostRuntimeChannelsAsync()`

Reply:
Correct ! We only want to start Language Worker Channels in "WebHost" Scope only when running in Azure - Placeholder Mode